woocommerce/woocommerce-ios
# Login: Implement steps for Jetpack setup

Part of #8075 


### Description

This PR implements the steps of the native Jetpack setup flow:
- Since there can be cases where the user dismisses the setup flow halfway through, we need to fetch the current status of their site's Jetpack before starting the setup. Based on the response, the first step is determined:
  - If the fetch fails, it may mean that the site doesn't have Jetpack, so we trigger installation.
  - If the fetch returns a Jetpack plugin with inactive status, we trigger activation.
  - If the returned Jetpack plugin is active, we trigger fetching connection URL.
- For each step, we trigger the next step when it completes successfully.
- When a Jetpack connection URL is fetched, we present a web view with that URL. I'm using the existing authenticated view controller and the Jetpack connection view model since the SwiftUI `AuthenticatedWebView` doesn't trigger exit with a given URL properly. Plus, we can reuse analytics in the existing screen.
- When Jetpack connection is authorized and the user taps the Go to Store button, the coordinator synchronizes data and triggers the site picker based on the matching site.

Error handling and tracks will be handled in future PRs.

### Testing instructions

- Log out of the app or skip login onboarding if needed.
- On the prologue screen, select "Log In" or "Continue with WordPress.com" based on the A/B test variant you get.
- Log in with your WordPress.com account.
- If your account doesn't have any associated site, notice that the empty store picker is displayed with the matching design as on Figma.
- Select Add a Store > Connect an existing site.
- Enter the address of your test store and tap Continue.
- On the Jetpack error screen, select Install Jetpack or Connect Jetpack.
- When the site credential login screen is presented, enter the correct credentials for your store.
- On the Jetpack setup screen, notice that the setup steps are handled subsequently. 
- When the setup reaches the connection step, notice that a web view is presented for approving Jetpack connection.
- Tap on the Authorize button on the web page, notice that the web view is then dismissed, and the Jetpack setup screens should then show that all steps are completed.
- Tap on Go to Store, notice that you are navigated to the home screen if your site has WooCommerce. Otherwise, the store picker should be displayed and your site has a ⚠️ icon - you should be able to install Woo from there.

### Screenshots


https://user-images.githubusercontent.com/5533851/203045798-adaf8b4a-817d-47ae-b240-99442b606957.mp4



---
